test/test_helper.rb

```diff
@@ -2,6 +2,7 @@
 require 'minitest/autorun'
 require 'minitest/pride'
 require 'mocha/mini_test'
+require 'timecop'
```

Comment:
Adding this just for temp dirs makes the tests a lot more complex... can you explain why we handle the urls in a temp dir the way we do? Its quite complex for something that shouldnt be so complex.